src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -327,12 +333,19 @@ Package url{6}
                     }
                     else
                     {
-                        this.Log().Info(logger, () => "{0}{1}".FormatWith(package.Identity.Id, config.ListCommand.IdOnly ? string.Empty : "|{0}".FormatWith(package.Identity.Version.ToFullStringChecked())));
+                        if (!(packageInfo != null && packageInfo.IsPinned && config.ListCommand.IgnorePinned))
+                        {
+                            this.Log().Info(logger, () => "{0}{1}".FormatWith(package.Identity.Id, config.ListCommand.IdOnly ? string.Empty : "|{0}".FormatWith(package.Identity.Version.ToFullStringChecked())));
+
+                        }
                     }
                 }
                 else
-                {
-                    this.Log().Debug(() => "{0}{1}".FormatWith(package.Identity.Id, config.ListCommand.IdOnly ? string.Empty : " {0}".FormatWith(package.Identity.Version.ToFullStringChecked())));
+                {   
+                    if (!(packageInfo != null && packageInfo.IsPinned && config.ListCommand.IgnorePinned))
+                    {
+                        this.Log().Debug(() => "{0}{1}".FormatWith(package.Identity.Id, config.ListCommand.IdOnly ? string.Empty : " {0}".FormatWith(package.Identity.Version.ToFullStringChecked())));
+                    }

```

Comment:
Why do we need this part of the if/else statement?

IF we go into this else statement, it would mean that `LocalOnly` is not requested, and there is no local metadata available.

Do we want to hide these packages even for remote searches?

Reply:
Looking at this again, I don't think that this if is needed.  So I will get it removed.